### PR TITLE
Refactoring creating and updating children

### PR DESCRIPTION
**Pull request by oleksandr-vovk**

### CODE REVIEW

**Review by alrom — APPROVED** [review]

File: OutOfSchool/OutOfSchool.WebApi/Models/Child/ChildBaseDTO.cs [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
-using System.Text.Json.Serialization;
 using OutOfSchool.Services.Enums;
 using OutOfSchool.WebApi.Models.SocialGroup;
 
 namespace OutOfSchool.WebApi.Models;
 
-public class ChildDto
+public class ChildBaseDto
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.WebApi/Models/Child/ChildBaseDTO.cs:7]: If i remember correctly, to create a child we don't need `List<SocialGroupDto>` but `List<long>` of social group Ids. Maybe move it out of base and have these lists in their own normal/create/update entities?

File: OutOfSchool/OutOfSchool.WebApi/Models/Child/ChildUpdateDto.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿namespace OutOfSchool.WebApi.Models;
+
+public class ChildUpdateDto : ChildBaseDto
+{
+    public Guid Id { get; set; }
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.WebApi/Models/Child/ChildUpdateDto.cs:5]: we can (and need) to put Id in the controller update path parameter

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChildController.cs [hunk 3]
```diff
@@ -235,7 +235,7 @@ public async Task<IActionResult> CreateChildren(List<ChildDto> childrenDtos)
     [ProducesResponseType(StatusCodes.Status403Forbidden)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
     [HttpPut]
-    public async Task<IActionResult> Update(ChildDto dto)
+    public async Task<IActionResult> Update(ChildUpdateDto dto)
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChildController.cs:238]: probably need to add Id as path parameter. Also probably the good time to make this more REST :) like (i will share excel file to Discord with more details)

get | /api/v1/Child/GetAllForAdmin | change | get | /api/v1/children
-- | -- | -- | -- | --
get | /api/v1/Child/GetChildrenListByParentId/{parentId}/{isParent} | change | get | /api/parents/{parentId}/children?{isParent=true/false}
get | /api/v1/Child/GetByParentIdForAdmin/{parentId} | change | get | TODO
get | /api/v1/Child/GetUsersChildren | change | get | /api/v1/children/my
get | /api/v1/Child/GetUsersChildById/{id} | change | get | /api/v1/children/my/{id}
get | /api/v1/Child/GetApprovedByWorkshopId/{workshopId} | change | get | TODO
post | /api/v1/Child/Create | change | post | /api/v1/children
put | /api/v1/Child/Update | change | put | /api/v1/children/{id}
delete | /api/v1/Child/Delete/{id} | change | delete | /api/v1/children/{id}

**Review by DmyMi — APPROVED** [review]

File: OutOfSchool/OutOfSchool.WebApi/Models/Child/ChildBaseDTO.cs (hunk 1 above)
Comment (oleksandr-vovk) [reply OutOfSchool/OutOfSchool.WebApi/Models/Child/ChildBaseDTO.cs:7]: done

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChildController.cs (hunk 3 above)
Comment (oleksandr-vovk) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChildController.cs]: done

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChildController.cs [hunk 4]
```diff
@@ -142,7 +142,7 @@ public async Task<IActionResult> GetUsersChildById(Guid id)
     [ProducesResponseType(StatusCodes.Status401Unauthorized)]
     [ProducesResponseType(StatusCodes.Status403Forbidden)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
-    [HttpGet("{workshopId}")]
+    [HttpGet("/api/v{version:apiVersion}/workshop/{id}/children/approved/")]
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChildController.cs:145]: should be `workshops`.

Also last `/` is not necessary.

Comment (oleksandr-vovk) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChildController.cs]: fixed

**Pull request merged**